File: ProspectManagerWebApi/Models/Evenement.cs
﻿using System.Text.Json.Serialization;

namespace ProspectManagerWebApi.Models
{
    public class Evenement : ITableHistorique
    {
        public int Id { get; set; }
        public TypeEvenement? TypeEvenement { get; set; }
        public DateTimeOffset DateEvenement {  get; set; }
        public string? Resultat { get; set; }
        public Contact? Contact { get; set; }
        public int Evaluation { get; set; }
        public ICollection<Produit>? Produits { get; set; }
        public Utilisateur? Utilisateur { get; set; }
        public ICollection<Modification> Modifications { get; set; } = new List<Modification>();
        [JsonIgnore]
        public Prospect Prospect { get; set; } = new Prospect();
    }
}

File: ProspectManagerClient/src/app/modules/gestion-prospect/models/evenement.ts
import { Contact } from "./contact";
import { Modification } from "./modification";
import { Produit } from "./produit";
import { TypeEvenement } from "./type-evenement";
import { Utilisateur } from "./utilisateur";

export class Evenement {
    id: number;
    typeEvenement: TypeEvenement;
    dateEvenement: Date;
    resultat: string;
    contact: Contact;
    produits: Produit[] = [];
    evaluation: number;
    utilisateur: Utilisateur;
    modifications: Modification[];
}